main.go

```diff
@@ -166,26 +183,52 @@ func mailHandler(peer smtpd.Peer, env smtpd.Envelope) error {
 
 	if err != nil {
 		var smtpError smtpd.Error
-		log.Printf("delivery failed: %v\n", err)
 
 		switch err.(type) {
 		case *textproto.Error:
 			err := err.(*textproto.Error)
 			smtpError = smtpd.Error{Code: err.Code, Message: err.Msg}
+
+			log.WithField("err_code", err.Code).
+				WithField("err_msg", err.Msg).
+				WithField("uuid", uniqueID).
```

Comment:
like the idea of uuid to tie together the logs for a request.

questions: 
- will it be indexed by loki? 
- if yes, how do we tell loki to not index it? 

seems like loki doesn't like high cardinality labels https://github.com/grafana/loki/issues/91

Reply:
No, it won't be indexed because of the issue you correctly pointed out
However, I've included the UUID in all log entries related to the mail send path, so if an error occurs in any step we can correlate all logs from that request by doing something like `{app="smtprelay"} | json | uuid="..."` to find all the additional details